src/main.rs

```diff
@@ -192,167 +185,190 @@ impl<'a> Interpreter<'a> {
                 ArrayGet => self.do_array_get(),
 
                 Call(ref target) => {
-                    let new_func = self.module.funcs.get(target).unwrap();
-                    let mut frame = CallFrame::for_fn(new_func, (func, pc));
+                    let new_func = self.module.funcs[target].clone();
```

Comment:
Eek, cloning the function is pretty heavyweight isn't it, given that we don't appear to be mutating it?